src/util.rs

```diff
@@ -81,9 +95,12 @@ fn decode_variable<R: Read>(reader: &mut R) -> Result<u64, Error> {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
 
     #[test]
     fn test_zigzag() {
-        // assert_eq!(zig_i32(42i32), zig_i64(42i64))
+        let mut a = Vec::new();
+        let mut b = Vec::new();
+        assert_eq!(zig_i32(42i32, &mut a), zig_i64(42i64, &mut b))
```

Comment:
`zig_i{32,64}` do not return anything. Instead they write into their `&mut buf`.
Your test ensures both functions return the same void 😛

(I won't say much, because I commented the test when I changed these functions to write into a buffer and forgot/was probably too lazy to test buffers had the same contents)